### PR TITLE
New article: Optimize binary search of Rust

### PR DESCRIPTION
**Pull request by Folyd**

### CODE REVIEW

**Review by dmosc — APPROVED** [review]
Super interesting read @Folyd. Very high-quality post! 👍

LGTM!

**Review by pau1tuck — COMMENTED** [review]
Fixed typos and capitalization

**Comment by Folyd** [comment]
> Fixed typos and capitalization

Thanks for your review. I guess you forgot to submit the comment?

File: content/issue-2/optimize-binary-search.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+  let mut a: usize = std::env::args().nth(1).unwrap().parse().unwrap_or_defaul();
+  if a > 42 {
+    a = 1;
+  } else {
+    a = 0;
+  }
+}
+```
+
+Here's the relevant portion of the resulting assembly code:
```
Comment (mfrw) [inline content/issue-2/optimize-binary-search.md:152]: Should we add a small pointer on how do we get to this assembly.
e.g what command to run etc etc
[btw, excellent and informative post so-far :)]

**Review by mfrw — COMMENTED** [review]
A very well written and informative post :)
Thank you @Folyd

File: content/issue-2/optimize-binary-search.md (hunk 1 above)
Comment (Folyd) [reply content/issue-2/optimize-binary-search.md:152]: Good point, I'll add a Godbolt link.

**Comment by pau1tuck** [comment]
> > Fixed typos and capitalization
> 
> Thanks for your review. I guess you forgot to submit the comment?

My apologies. Superb high-quality article. Very impressive. 👍

**Pull request merged**